tools/syz-trace2syz/proggen/proggen_test.go

```diff
@@ -0,0 +1,320 @@
+package proggen
+
+import (
+	"fmt"
+	"github.com/google/syzkaller/pkg/log"
+	"github.com/google/syzkaller/prog"
+	_ "github.com/google/syzkaller/sys"
+	"github.com/google/syzkaller/tools/syz-trace2syz/parser"
+	"strings"
+	"testing"
+)
+
+var (
+	OS   = "linux"
+	Arch = "amd64"
+)
+
+func parseSingleTrace(t *testing.T, data string) *Context {
+	var err error
+	var target *prog.Target
+	var traceTree *parser.TraceTree
+	var ctx *Context
+
+	target, err = prog.GetTarget(OS, Arch)
+	if err != nil {
+		t.Fatalf("Failed to load target %s\n", err)
+	}
+	variantMap := NewCall2VariantMap()
+	variantMap.Build(target)
+	traceTree = parser.ParseLoop(data)
+	ctx = GenSyzProg(traceTree.TraceMap[traceTree.RootPid], target, variantMap)
+
+	ctx.FillOutMemory()
+	if err = ctx.Prog.Validate(); err != nil {
+		t.Fatalf("Failed to parse trace: %s", err.Error())
+	}
+	return ctx
+}
+
+func testMatchingCallSequence(calls []*prog.Call, seq []string) error {
```

Comment:
prog.P has String method which returns concatenation of all calls.
So you can encode the expected result as "mmap-open-connect" and then just compare this with p.String().